File: source/SampleNet4VB/Code/FastStubbornDirectoryCleaner.vb

Namespace Portal_Convert.CdpConverter


    Public Class FastStubbornDirectoryCleaner
        Implements MasterDevs.ChromeDevTools.IDirectoryCleaner


        Public Sub Delete(ByVal dir As System.IO.DirectoryInfo) Implements MasterDevs.ChromeDevTools.IDirectoryCleaner.Delete

            Dim t As System.Threading.Tasks.Task(Of Integer) = System.Threading.Tasks.Task2.Run(
                Function()

                    While True

                        Try
                            dir.Delete(True)
                            Return 0
                        Catch ex As System.Exception

                            Try
                                Dim fis As System.IO.FileInfo() = dir.GetFiles("*.*", System.IO.SearchOption.AllDirectories)

                                For Each fi As System.IO.FileInfo In fis
                                    System.IO.File.SetAttributes(fi.FullName, System.IO.FileAttributes.Normal)
                                Next

                            Catch ex2 As System.Exception
                                System.Console.WriteLine(System.Environment.NewLine)
                                System.Console.WriteLine("Error removing write-protection:")
                                System.Console.WriteLine(ex2.Message)
                            End Try

                            System.Console.WriteLine(ex.Message)
                            System.Threading.Thread.Sleep(500)
                            System.Console.WriteLine("Repeat")
                        End Try
                    End While

                    Return 0
                End Function)
        End Sub


    End Class


End Namespace

File: source/SampleNet4/Code/FastStubbornDirectoryCleaner.cs

namespace Portal_Convert.CdpConverter
{


    public class FastStubbornDirectoryCleaner
        : MasterDevs.ChromeDevTools.IDirectoryCleaner
    {

        public void Delete(System.IO.DirectoryInfo dir)
        {
            System.Threading.Tasks.Task<int> t = System.Threading.Tasks.Task2.Run(delegate ()
            {
                while (true)
                {
                    try
                    {
                        dir.Delete(true);
                        return 0;
                    }
                    catch (System.Exception ex)
                    {
                        try
                        {
                            System.IO.FileInfo[] fis = dir.GetFiles("*.*", System.IO.SearchOption.AllDirectories);

                            foreach (System.IO.FileInfo fi in fis)
                            {
                                System.IO.File.SetAttributes(fi.FullName, System.IO.FileAttributes.Normal);
                            }
                        }
                        catch (System.Exception ex2)
                        {
                            System.Console.WriteLine(System.Environment.NewLine);
                            System.Console.WriteLine("Error removing write-protection:");
                            System.Console.WriteLine(ex2.Message);
                        }

                        System.Console.WriteLine(ex.Message);
                        System.Threading.Thread.Sleep(500);
                        System.Console.WriteLine("Repeat");
                    }
                }

            }
            );

        } // End Sub Delete 


    } // End Class DiagnosticDirectoryCleaner 


} // End Namespace Portal_Convert.CdpConverter
